packages/grafana-ui/src/components/Icon/Icon.tsx

```diff
@@ -1,11 +1,18 @@
-import React from 'react';
-import { cx, css } from 'emotion';
-import { stylesFactory } from '../../themes';
+import React, { PureComponent, ComponentType } from 'react';
+import { css, cx } from 'emotion';
+
+import { stylesFactory, withTheme } from '../../themes';
+import { Themeable } from '../../types';
 import { IconType } from './types';
 
-export interface IconProps {
+type Size = 'xs' | 'sm' | 'base' | 'md' | 'lg';
+
+interface IconProps extends Themeable {
```

Comment:
wonder if can use something like  make it extend HtmlElementAttributes or something so you do not need to add className or onClick onMouseDown etc, 

Also maybe consider adding React.forwardRef

Reply:
yes, agree on this